src/adapters/wallet.js

```diff
@@ -181,6 +201,73 @@ class WalletAdapter {
     }
   }
 
+  // Create an instance of minimal-avax-wallet. Use data in the wallet-avax.json file
+  async instanceAvaxWallet (walletData) {
+    try {
+      if (typeof walletData !== 'object') {
+        throw new Error('walletData must be an object with the wallet information')
+      }
+
+      // Instantiate minimal-avax-wallet.
+      this.avaxWallet = new this.AvaxWallet(walletData.mnemonic)
+
+      // Wait for wallet to initialize.
+      await this.avaxWallet.walletInfoPromise
+
+      return true
+    } catch (err) {
+      console.error('Error in instanceAvaxWallet()')
+      throw err
+    }
+  }
+
+  // Make offfer Tx
+  async createPartialTxHex (avaxAmount, privateKey) {
+    let tokenWallet = this.avaxWallet
+
+    if (privateKey) {
+      tokenWallet = new this.AvaxWallet(privateKey)
+      await tokenWallet.walletInfoPromise
+    }
+
+    // arrange the inputs
+    const addrReferences = {}
+    const inputs = []
+    const address = tokenWallet.walletInfo.address
+
+    for (const item of tokenWallet.utxos.utxoStore) {
+      const utxo = tokenWallet.utxos.encodeUtxo(item, address)
+      const utxoID = utxo.getUTXOID()
+
+      addrReferences[utxoID] = address
+      inputs.push(utxo)
+    }
```

Comment:
This for loop should probably have a conditional to skip utxos, but since this method is expected to be use only with newly created address with a single UTXO, the condition is most likely superfluous. It might even be changed to just:
`
cosnt tokenInput = tokenWallet.utxos.encodeUtxo(tokenWallet.utxos.utxoStore[0], address)
cosnt inputs = [tokenInput]
`

Reply:
I think that simplification is a fine way to go. At this point, I believe the assumption is correct: We're only going to be dealing with a single UTXO.